devicetypes/sky-nie/minoston-wallmote.src/minoston-wallmote.groovy

```diff
@@ -0,0 +1,318 @@
+/**
+ *  Minoston Wallmote v1.0.1
+ *
+ *  	Models: Eva Logik (ZW924) / MINOSTON (MR40Z)
+ *
+ *  Author:
+ *   winnie (sky-nie)
+ *
+ *	Documentation:
+ *
+ *  Changelog:
+ *
+ *    1.0.1 (07/07/2021)
+ *      - Initial Release
+ *
+ * Reference：
+ *  https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy
+ *
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+metadata {
+    definition (name: "Minoston Wallmote", namespace: "sky-nie", author: "winnie", ocfDeviceType: "x.com.st.d.remotecontroller", mcdSync: true) {
+        capability "Actuator"
+        capability "Button"
+        capability "Battery"
+        capability "Configuration"
+        capability "Sensor"
+        capability "Health Check"
+
+        fingerprint mfr: "0312", prod: "0924", model: "D001", deviceJoinName: "Minoston Remote Control", mnmn: "SmartThings", vid: "generic-4-button"//ZW924
+    }
+
+    tiles(scale: 2) {
+        multiAttributeTile(name: "rich-control", type: "generic", width: 6, height: 4, canChangeIcon: true) {
+            tileAttribute("device.button", key: "PRIMARY_CONTROL") {
+                attributeState "default", label: ' ', action: "", icon: "st.unknown.zwave.remote-controller", backgroundColor: "#ffffff"
+            }
+        }
+        valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "battery", label:'${currentValue}% battery', unit:""
+        }
+
+        main("rich-control")
+        details(["rich-control", childDeviceTiles("endpoints"), "battery"])
+    }
+
+    preferences {
+        configParams.each {
+            if (it.name) {
+                if (it.range) {
+                    getNumberInput(it)
+                }
+                else {
+                    getOptionsInput(it)
+                }
+            }
+        }
+    }
+}
+
+def getNumberOfButtons() {
+    def modelToButtons = ["D001":4, "0082" : 4, "0081": 2, "0003": 2]
+    return modelToButtons[zwaveInfo.model] ?: 1
+}
```

Comment:
Why do you need this method, when only one fingerprint is added to the DTH?

Reply:
I deleted irrelevant code, but I kept this function because I am not sure if similar products will be added later.

Reply:
If similar products will be added later then this method shouldn't be here now.